src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -479,10 +479,13 @@ public override string ToString()
         /// <summary>
         /// Fires when a Command (e.g. Enable) comes from a an EventListener.  
         /// </summary>
-        public event EventHandler<EventCommandEventArgs> EventCommandExecuted // TODO-NULLABLE: Should all events use nullable delegate types?
+        public event EventHandler<EventCommandEventArgs>? EventCommandExecuted
```

Comment:
@billwert, @brianrob, @noahfalk, could I get a code review of these EventSource tweaks, in particular the changes here and in CallBackForExistingEventSources, which skip code if a null callback is passed.  In addition to adding the appropriate nullable annotations, this prevents code like this from null ref'ing:
```C#
using System.Diagnostics.Tracing;
using System.Threading.Tasks;

class Program
{
    static void Main()
    {
        Task.Run(() => { });
        var listener = new MyEventListener();
        listener.EventSourceCreated += null;
    }
}

internal class MyEventListener : EventListener
{
    protected override void OnEventSourceCreated(EventSource eventSource) =>
        EnableEvents(eventSource, EventLevel.LogAlways);
}
```
where today that'll null ref:
```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Diagnostics.Tracing.EventListener.CallBackForExistingEventSources(Boolean addToListenersList, EventHandler`1 callback) in D:\repos\coreclr\src\System.Private.CoreLib\shared\System\Diagnostics\Tracing\EventSource.cs:line 4530
   at System.Diagnostics.Tracing.EventListener.add_EventSourceCreated(EventHandler`1 value) in D:\repos\coreclr\src\System.Private.CoreLib\shared\System\Diagnostics\Tracing\EventSource.cs:line 4049
   at Program.Main() in C:\Users\stoub\source\repos\ConsoleApp13\ConsoleApp13\Program.cs:line 10
```

Reply:
@stephentoub, the EventSource changes seem reasonable to me.  Thanks.

---
In reply to: [304586154](https://github.com/dotnet/coreclr/pull/25752#discussion_r304586154) [](ancestors = 304586154)

Reply:
Thanks.

Reply:
LGTM as well

Reply:
Thanks